src/App.jsx

```diff
@@ -1,27 +1,52 @@
 import './App.css';
-import React from 'react';
-
-import RegisterLoginModal from '/src/components/RegisterLoginModal/RegisterLoginModal';
 
 import { ThemeProvider } from '@mui/material/styles';
+import React, { useCallback, useEffect, useState } from 'react';
 import { BrowserRouter, Route, Routes } from 'react-router-dom';
 
 import SellHousePage from '/src/pages/SellHousePage/SellHousePage';
 import theme from '/src/theme/theme';
 
+import { readAll } from './utils/api';
+
 function App() {
+  const [houses, setHouses] = useState(null);
+  const [error, setError] = useState(false);
+  const [loading, setLoading] = useState(true);
```

Comment:
It's a very good idea to use it here. You could also consider using `React Query` to do so.

Reply:
Thanks. I will read about it :)